src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  actions.forEach((item, index, array) => {
+    let newObj;
+
+    switch (item.type) {
+      case 'addProperties':
+        newObj = result.length === 0
+          ? { ...state }
+          : { ...result[result.length - 1] };
```

Comment:
You are repeating this block of code in each case, consider moving this check upper in your scope